test (#50)
shedding-hub/shedding-hub: data/kimse2020viral/kimse2020viral-extraction 16.12.40.md

Reviewer: Remove the "16.12.40" in the file name.

# Extraction for kimse et al. (2020)

> [kimse et al. (2020)](https://www.ijidonline.com/article/S1201-9712(20)30299-X/fulltext) The authors measured SARS-CoV-2 in longitudinal throat swab samples collected from 71 COVID-19 patients between February 4 and April 7, 2020. Abundances were quantified using real-time reverse transcription polymerase chain reaction (RT-PCR). Specimens were collected from all patients at least 2 days after hospitalization and physicians checked their symptoms and signs daily. Patients who had asymptomatic carrier and incubation period were analyzed. The raw data is stored at [Shedding Hub](https://github.com/shedding-hub). 

Reviewer: ```suggestion
[Kim et al. (2020)](https://www.ijidonline.com/article/S1201-9712(20)30299-X/fulltext) The authors measured SARS-CoV-2 in longitudinal throat swab samples collected from 71 COVID-19 patients between February 4 and April 7, 2020. Abundances were quantified using real-time reverse transcription polymerase chain reaction (RT-PCR). Specimens were collected from all patients at least two days after hospitalization and physicians checked their symptoms and signs daily. Patients who had asymptomatic carrier and incubation period were analyzed. The raw data is stored at [Shedding Hub](https://github.com/shedding-hub). 
```

First, we `import` python modules needed:

```python
import yaml
import pandas as pd

> import numpy as np

Reviewer: Did you use numpy? if you did not use it, just delete this line.

```

```python
participants_symptomatic = []

for patient_id, patient_data in combineddataset_df.groupby("PatientID"):
    participant = {
        "attributes": {
            "age": float(patient_data["Age"].iloc[0]),
            "sex": str(patient_data["Sex"].iloc[0]),
        },
        "measurements": []
    }
    for _, row in patient_data.iterrows():
        measurement = {
            "analyte": "throatswab_SARSCoV2",
            "time": max(int(row["Day"]), 1), 
            "value": float(row["value"]) if float(row["value"]) != 1.0 else 'negative'
        }
        participant["measurements"].append(measurement)
    
    participants_symptomatic.append(participant)

```

```python
participants_asymptomatic = []
for patient_id, patient_data in asymptomatic_df.groupby("PatientID"):
    participant = {
        "attributes": {
            "age": float(patient_data["Age"].iloc[0]),
            "sex": str(patient_data["Sex"].iloc[0]),
        },
        "measurements": []
    }
    for _, row in patient_data.iterrows():
        measurement = {
            "analyte": "throatswab_SARSCoV2",
            "time": max(int(row["Day"]), 1), 

>             "value": float(row["value"]) if float(row["value"]) != 1.0 else 'negative'

Reviewer: For those 3 symptomatic cases, use the CT value, `Ctvalue` instead of viral load `value` in the CombinedDataset.xlsx.

        }
        participant["measurements"].append(measurement)
    
    participants_asymptomatic.append(participant)
```

```python
participants = []
participants.extend(participants_symptomatic)
participants.extend(participants_asymptomatic)
```

Finally, the data is formatted and output as a YAML file.
```python
output_data = {
    "title": "Viral kinetics of SARS-CoV-2 in asymptomatic carriers and presymptomatic patients",
    "doi": "10.1016/j.ijid.2020.04.083",
    "description": folded_str(

>         "The authors measured SARS-CoV-2 in longitudinal throat swab samples collected from 71 COVID-19 patients between February 4 and April 7, 2020..."

Reviewer: ```suggestion
        "The authors measured SARS-CoV-2 in longitudinal throat swab samples collected from 71 COVID-19 patients between February 4 and April 7, 2020.\n"
```

    ),
    "analytes": {
        "throatswab_SARSCoV2_symptomatic": {

>             "description": folded_str("SARS-CoV-2 RNA genome copy concentration in throat swab samples..."),

Reviewer: Add this to the description "Ct = 35 is the cut-off for a positive result and Ct = 40 is a negative sample; Ct = 40 was the limit of detection."

            "specimen": "throat_swab",
            "biomarker": "SARS-CoV-2",
            "gene_target": "RdRp",
            "limit_of_quantification": "unknown",
            "limit_of_detection": 40,

>             "unit": "gc/swab",

Reviewer: ```suggestion
            "unit": "cycle threshold",
```

            "reference_event": "symptom onset"
        },
         "throatswab_SARSCoV2_asymptomatic": {

>             "description": folded_str("SARS-CoV-2 RNA genome copy concentration in throat swab samples..."),

Reviewer: Add this to the description "Ct = 35 is the cut-off for a positive result and Ct = 40 is a negative sample; Ct = 40 was the limit of detection."

            "specimen": "throat_swab",
            "biomarker": "SARS-CoV-2",
            "gene_target": "RdRp",
            "limit_of_quantification": "unknown",
            "limit_of_detection": 40,

>             "unit": "gc/swab",

Reviewer: ```suggestion
            "unit": "cycle threshold",
```
